code/modules/mob/living/carbon/human/species/golem.dm

```diff
@@ -357,10 +373,11 @@
 
 /datum/species/golem/uranium/handle_life(mob/living/carbon/human/H)
 	for(var/mob/living/L in range(2, H))
-		if(isgolem(H))
-			continue
-		to_chat(L, "<span class='danger'>You are enveloped by a soft green glow emanating from [H].</span>")
-		L.apply_effect(10, IRRADIATE)
+		if(ishuman(L))
+			var/mob/living/carbon/human/I = L
+			if(!(RADIMMUNE in I.dna.species.species_traits))
```

Comment:
No need for the RADIMMUNE check here I believe, as mobs who are RADIMMUNE still soak up radiation. They just don't suffer any negative consequences from it.

Reply:
Yep, that's absolutely correct

Reply:
Not having the check will cause the chat to be spammed with 'You are envelopped..' for everybody nearby, which is annoying. A check is necessary in any case so that the uranium golem himself isnt spammed by it.

Reply:
You can just continue the loop if it’s on the source golem, that way it won’t spam them. And won’t the spam be an issue for non-immune species as well in that case?

Reply:
It's not really an issue because they arent immune to rads so they need to get the fuck out. I'll add a prob check to reduce the amount of messages.